packages/grid/_modules_/grid/components/panel/GridPanel.tsx

```diff
@@ -23,11 +23,12 @@ export interface GridPanelProps extends StandardProps<PopperProps, 'children'> {
    */
   classes?: Partial<GridPanelClasses>;
   open: boolean;
+  as: React.ElementType;
 }
 
 export const gridPanelClasses = generateUtilityClasses('MuiDataGrid', ['panel', 'paper']);
 
-const GridPanelRoot = styled(Popper, {
+const GridPanelRoot = styled('div', {
```

Comment:
Although I suggested it, I don't know if this is the right approach. Looking at the change done in https://github.com/mui-org/material-ui-x/pull/3490/files#diff-c2fd479a822e3ad62e4338f61294f31a114e307bcbcce46c482822ef9d8fc784 maybe the default tag should be `Popper` and keep the `as` prop too. 

Looking closer I found a similar case in the core:

https://github.com/mui-org/material-ui/blob/bc016a59f8b4afeb43c98f677c9fbd1e8afb2698/packages/mui-material/src/Autocomplete/Autocomplete.js#L408

https://github.com/mui-org/material-ui/blob/bc016a59f8b4afeb43c98f677c9fbd1e8afb2698/packages/mui-material/src/Autocomplete/Autocomplete.js#L581-L582

The default value of `PopperComponent` is `Popper`.

Reply:
Updated